drivers/tee/optee/optee.c

```diff
@@ -434,15 +437,127 @@ static int optee_invoke_func(const struct device *dev, struct tee_invoke_func_ar
 	return (rc) ? rc : ret;
 }
 
-static int optee_shm_register(const struct device *dev, void *addr, uint64_t size, uint32_t flags,
-			      struct tee_shm *shm)
+static void *optee_construct_page_list(void *buf, uint32_t len, uint64_t *phys_buf)
 {
-	return 0;
+	const uint32_t page_size = OPTEE_MSG_NONCONTIG_PAGE_SIZE;
+	const uint32_t num_pages_in_pl = page_size - 1;
```

Comment:
This is wrong. It should be `OPTEE_NUMBER_OF_ADDR_PER_PAGE - 1`

Reply:
+